lib/wibox/layout/margin.lua

```diff
@@ -23,7 +23,7 @@ function margin:draw(wibox, cr, width, height)
     local h = self.bottom
     local color = self.color
 
-    if not self.widget or width <= x + w or height <= y + h then
+    if not self.widget or not self.widget.visible or width <= x + w or height <= y + h then
```

Comment:
I'd drop this as well. Otherwise a margin around a widget of size `0,0` would look different from a margin around a widget with `visible = true` and that seems like a weird thing to me.